ztp/ztp-policy-generator/kustomize/plugin/policyGenerator/v1/policygenerator/policyGen/policyBuilder.go

```diff
@@ -36,83 +42,85 @@ func (pbuilder *PolicyBuilder) Build(customResourseOnly bool) (map[string]interf
 			if err != nil {
 				panic(err)
 			}
-			_, resourceDef := pbuilder.getCustomResource(sFile.Data, sFile.Spec, sFile.Labels,sPolicyFile, rname, pbuilder.PolicyGenTemp.Metadata.Labels.Mcp)
-
-			acmPolicy := pbuilder.getPolicy( name, namespace, resourceDef)
+			resourcesDef := pbuilder.getCustomResources(sFile, sPolicyFile)
+			acmPolicy := pbuilder.getPolicy( name, namespace, resourcesDef)
 			policies[path + "/" + name] = acmPolicy
 			subject := CreatePolicySubject(name)
 			subjects = append(subjects, subject)
 		}
-
 		placementRule := CreatePlacementRule(pbuilder.PolicyGenTemp.Metadata.Name, namespace, matchKey, matchOper, matchValue)
 		err := CheckNameLength(namespace, placementRule.Metadata.Name)
+
 		if err != nil {
 			panic(err)
 		}
 		policies[path + "/" + placementRule.Metadata.Name] = placementRule
-
 		placementBinding := CreatePlacementBinding(pbuilder.PolicyGenTemp.Metadata.Name, namespace, placementRule.Metadata.Name, subjects)
 		err = CheckNameLength(namespace, placementBinding.Metadata.Name)
+
 		if err != nil {
 			panic(err)
 		}
 		policies[path + "/" + placementBinding.Metadata.Name] = placementBinding
 	} else if len(pbuilder.PolicyGenTemp.SourceFiles) != 0 && customResourseOnly {
-		for id, sFile := range pbuilder.PolicyGenTemp.SourceFiles {
-			_, rname := pbuilder.getPolicyName(id)
+		for _, sFile := range pbuilder.PolicyGenTemp.SourceFiles {
 			sPolicyFile, err := ioutil.ReadFile(pbuilder.SourcePoliciesDir + "/" + sFile.FileName + utils.FileExt)
+
```

Comment:
There seems to be enough common logic between the if block above and this else-if that you could have them structured as:
```GO
if len(pbuilder.PolicyGenTemp.SourceFiles) != 0 {
    for ... sourceFiles {
        sPolicyFile := ....
        resources := pbuilder.getCustomResources.....
        if customResourceOnly {
             <logic from here>
        } else {
            <logic from above>
        }
    }
}
```